pkg/cleanup/types/built_in.go

```diff
@@ -63,6 +64,26 @@ func serviceCleanup(spec, actual *unstructured.Unstructured) (*unstructured.Unst
 	return util.RuntimeToUnstructured(&serviceSpec)
 }
 
+func secretCleanup(spec, actual *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	var secretSpec core_v1.Secret
+	if err := unstructured_conversion.DefaultConverter.FromUnstructured(spec.Object, &secretSpec); err != nil {
+		return nil, err
+	}
+
+	// StringData overwrites Data
+	if len(secretSpec.StringData) > 0 {
+		if secretSpec.Data == nil {
+			secretSpec.Data = make(map[string][]byte, len(secretSpec.StringData))
+		}
+		for k, v := range secretSpec.StringData {
+			secretSpec.Data[k] = []byte(v)
+		}
+		secretSpec.StringData = nil
+	}
+
+	return util.RuntimeToUnstructured(&secretSpec)
```

Comment:
So at this stage the `spec` object will have both `StringData` and `Data` set.
But the `actual` object will only have `Data` set, right? (`StringData` is write-only field). Do you set it somewhere as well, or have some custom comparison logic?

Reply:
Line 81 cleans up `StringData` field.